### PR TITLE
Enhance `X509_REQ_get_extensions()`, add its doc, and improve `x509_att.c`

### PR DESCRIPTION
**Pull request by DDvO**
* `X509_REQ_get_extensions()`: add error queue entry on ill-formed extensions attribute
* add missing doc of `X509_REQ_get_extensions()` and `X509_REQ_add_extensions{,_nid}()`

On this occasion, improve error checking and reporting and coding style in `x509_att.c`.
In particular, there should be only one attribute per OID/NID, which is now checked.

### CODE REVIEW

**Review by t8m — APPROVED** [review]

**Comment by DDvO** [comment]
Also here, thanks @t8m for the OTC review,
and asking @openssl/committers for 2nd review.

File: doc/man3/X509_REQ_get_extensions.pod [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+This function must not be called more than once on the same I<req>.
+
+X509_REQ_add_extensions_nid() is like X509_REQ_add_extensions()
+except that I<nid> is used to identify the extensions attribute.
+This function must not be called more than once with the same I<req> and I<nid>.
+
+=head1 RETURN VALUES
+
+X509_REQ_get_extensions() returns a pointer to B<STACK_OF(X509_EXTENSION)>
+or NULL no error.
```
Comment (mattcaswell) [inline doc/man3/X509_REQ_get_extensions.pod:36]: typo: on error

**Review by mattcaswell — APPROVED** [review]
Approved subject to the minor typo being fixed.

File: doc/man3/X509_REQ_get_extensions.pod (hunk 1 above)
Comment (DDvO) [reply doc/man3/X509_REQ_get_extensions.pod:36]: Oops, fixed.

**Comment by t8m** [comment]
Still OK with the typo fixed.

**Comment by DDvO** [comment]
Merged - thanks @t8m and @mattcaswell

**Pull request closed without merging**